static/coffee/controllers.coffee

```diff
@@ -604,6 +714,11 @@ controllers.controller('CaseController', ['$scope', '$window', '$timeout', 'Case
       )
     )
 
+  # unlock message on exit
+  $window.onbeforeunload = ->
+    MessageService.checkLock($scope.caseObj.msg_backend_id, true)
```

Comment:
I'm not convinced this is useful - either it should be locking all messages that belong to a case or none of them - and I think it's ok to not lock them whilst a case is being viewed since the user can see that those messages are associated with a case anyway - and they've been archived.